wallet/src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -214,3 +223,50 @@ sendMoney SendActions{..} passphrase moneySource dstDistr = do
 
     logDebug "sendMoney: constructing response"
     fst <$> constructCTx srcWallet srcWalletAddrs diff th
+
+
+-- | For each specified transaction, if it has been canceled, creates
+-- new transaction with correspondent destination address and money.
+-- This works as `mapM newPayment`, so completion may take a lot of time.
+reformCanceledTxs
+    :: MonadWalletWebMode m
+    => SendActions m -> PassPhrase -> ReformCanceledTxsParams -> m [CTx]
+reformCanceledTxs sendActions passphrase params = do
+    ptxs <- getPendingTxs
+    let canceledTxs :: [(TxId, (CId Wal, PtxPoolInfo))]
+        canceledTxs = flip mapMaybe ptxs $ \PendingTx{..} ->
+            case _ptxCond of
+                PtxWontApply _ poolInfo -> Just (_ptxTxId, (_ptxWallet, poolInfo))
+                _                       -> Nothing
+    logDebug $ sformat ("List of all canceled transactions ever: "%listJson)
+         (map fst canceledTxs)
+
+    txsToRecreate <- case rctpBanned params of
+        [] -> pure canceledTxs
+        bannedCTxs -> do
+            bannedTxs <- mapM decodeCTypeOrFail bannedCTxs
+            let bannedTxsSet = S.fromList bannedTxs
+            pure $ filter ((`S.notMember` bannedTxsSet) . fst) canceledTxs
```

Comment:
Maybe it's overkill for the sake of this patch, but looks like we might benefit from converting `canceledTxs` and `bannedTxs` into a `Set` immediately and the use `Set` operations all along the way, when possible (unless there is a missing `Ord` constraint for those datatypes which doesn't allow us conversion).

Reply:
Is it because difference on sets works for `~O(n+m)`?

Reply:
I don't see how can I do it with `Set`s, because of `fst` there, should I use `Map`s then?